File: src/main/java/aurocosh/divinefavor/common/custom_data/living/data/cripple/CrippleData.java
package aurocosh.divinefavor.common.custom_data.living.data.cripple;

import aurocosh.divinefavor.common.config.common.ConfigArrow;
import aurocosh.divinefavor.common.lib.SimpleCounter;

public class CrippleData {
    private SimpleCounter counter;

    public CrippleData() {
        this.counter = new SimpleCounter(ConfigArrow.cripple.cureRate);
    }

    public void resetCureTimer() {
        counter.reset();
    }

    public boolean cureTick() {
        return counter.count();
    }

    public int getCureTicks() {
        return counter.getCount();
    }

    public void setCureTicks(int ticks) {
        counter.setCount(ticks);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/living/data/cripple/CrippleData.kt
package aurocosh.divinefavor.common.custom_data.living.data.cripple

import aurocosh.divinefavor.common.config.common.ConfigCurses
import aurocosh.divinefavor.common.lib.SimpleCounter

class CrippleData {
    private val counter: SimpleCounter = SimpleCounter(ConfigCurses.cripple.cureRate)

    var cureTicks: Int
        get() = counter.count
        set(ticks) {
            counter.count = ticks
        }

    fun resetCureTimer() {
        counter.reset()
    }

    fun cureTick(): Boolean {
        return counter.count()
    }
}
